src/librustc/ty/sty.rs

```diff
@@ -276,6 +279,50 @@ impl<'a, 'gcx, 'acx, 'tcx> ClosureSubsts<'tcx> {
     }
 }
 
+impl<'a, 'gcx, 'tcx> ClosureSubsts<'tcx> {
+    pub fn state_tys(self, def_id: DefId, tcx: TyCtxt<'a, 'gcx, 'tcx>) ->
```

Comment:
I think this function could use a comment. I am guessing it returns the types of local variables that are (potentially) saved/restore in the state?